credentials.go

```diff
@@ -10,14 +10,79 @@ import (
 	"fmt"
 	"net"
 	"net/url"
+	"strings"
 
 	"github.com/saucelabs/forwarder/log"
 )
 
+type Userinfo struct {
+	Username string
+	Password string
+}
+
+// ParseUserinfo parses a user:password string into Userinfo.
+// Username and password cannot be empty.
+func ParseUserinfo(val string) (*Userinfo, error) {
```

Comment:
Maybe we can change the API to use `Userinfo` not a pointer - this type is just two strings.
It would never escape and the API would not allow modifications by default.